org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/AcquisitionDevice.java

```diff
@@ -226,7 +226,7 @@ private void setScannables(ScanModel model) throws ScanningException {
 
 	private Set<String> getAxesToMove(IMalcolmDevice<?> malcolmDevice) {
 		try {
-			return ((IMalcolmDevice<?>) malcolmDevice).getAxesToMove();
+			return malcolmDevice.getAxesToMove();
 		} catch (ScanningException e) {
 			throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)